cli/cmd/cluster.go

```diff
@@ -533,6 +543,125 @@ var _downCmd = &cobra.Command{
 	},
 }
 
+var _exportCmd = &cobra.Command{
+	Use:   "export",
+	Short: "export all APIs deployed in a cluster",
+	Args:  cobra.NoArgs,
+	Run: func(cmd *cobra.Command, args []string) {
+		telemetry.Event("cli.cluster.export")
+
+		if _flagClusterConfig != "" {
+			// Deprecation: specifying aws creds in cluster configuration is no longer supported
+			if err := detectAWSCredsInConfigFile(cmd.Use, _flagClusterConfig); err != nil {
+				exit.Error(err)
+			}
+		}
+
+		accessConfig, err := getClusterAccessConfig(_flagClusterDisallowPrompt)
+		if err != nil {
+			exit.Error(err)
+		}
+
+		awsCreds, err := awsCredentialsForManagingCluster(*accessConfig, _flagClusterDisallowPrompt)
+		if err != nil {
+			exit.Error(err)
+		}
+
+		// Check AWS access
+		awsClient, err := newAWSClient(*accessConfig.Region, awsCreds)
+		if err != nil {
+			exit.Error(err)
+		}
+		warnIfNotAdmin(awsClient)
+
+		loadBalancer, err := awsClient.FindLoadBalancer(map[string]string{
+			clusterconfig.ClusterNameTag: *accessConfig.ClusterName,
+			"cortex.dev/load-balancer":   "operator",
+		})
+		if err != nil {
+			exit.Error(err)
+		}
+
+		operatorConfig := cluster.OperatorConfig{
+			Telemetry:          isTelemetryEnabled(),
+			ClientID:           clientID(),
+			AWSAccessKeyID:     awsCreds.AWSAccessKeyID,
+			AWSSecretAccessKey: awsCreds.AWSSecretAccessKey,
+			OperatorEndpoint:   "https://" + *loadBalancer.DNSName,
+		}
+
+		info, err := cluster.Info(operatorConfig)
+		if err != nil {
+			exit.Error(err)
+		}
+
+		apisResponse, err := cluster.GetAPIs(operatorConfig)
+		if err != nil {
+			exit.Error(err)
+		}
+
+		apiSpecList := []spec.API{}
+
+		for _, batchAPI := range apisResponse.BatchAPIs {
+			apiSpecList = append(apiSpecList, batchAPI.Spec)
+		}
+
+		for _, realtimeAPI := range apisResponse.RealtimeAPIs {
+			apiSpecList = append(apiSpecList, realtimeAPI.Spec)
+		}
+
+		for _, trafficSplitter := range apisResponse.TrafficSplitters {
+			apiSpecList = append(apiSpecList, trafficSplitter.Spec)
+		}
+
+		if len(apiSpecList) == 0 {
+			fmt.Println(fmt.Sprintf("no apis found in cluster named %s in %s", *accessConfig.ClusterName, *accessConfig.Region))
+			exit.Ok()
+		}
+
+		exportPath := "export"
+
+		_, err = files.CreateDirIfMissing(exportPath)
+		if err != nil {
+			exit.Error(err)
+		}
+
+		for _, apiSpec := range apiSpecList {
+			baseDir := filepath.Join(exportPath, apiSpec.Name+"-"+apiSpec.ID)
+
+			fmt.Println(fmt.Sprintf("exporting %s to %s", apiSpec.Name, baseDir))
+
+			err = os.Mkdir(baseDir, os.ModePerm)
```

Comment:
nit: use `files.CreateDir()`. Also note that this will error if it already exists, which I think is a good thing